app/components/topic-card/component.js

```diff
@@ -28,13 +29,25 @@ export default Ember.Component.extend({
       }
       this.get('topic').save();
     },
-    volunteer() {
-      if(this.get('isVolunteer')) {
-        this.get('topic.volunteers').removeObject(this.get('session.currentUser'));
-      }
-      else {
-        this.get('topic.volunteers').pushObject(this.get('session.currentUser'));
+    chooseTalkDate(date) {
+      // we don't currently have a date and the meetup is today.
+      // let them choose a date.
+      if (!date && this.get('meetupDates.isMeetupToday')) {
+        return this.toggleProperty('showMeetupDates');
       }
+
+      this.get('topic').setProperties({
+        talkBy: this.get('session.currentUser'),
+        talkDate: date
+      });
+
+      this.get('topic').save()
+        .then(() => {
+          this.set('showMeetupDates', false);
```

Comment:
I think I would be optimistic here and set this property immediately rather than putting it in a `.then()` callback, just in case it takes a second to come back then it won't feel laggy.